### DISCUSSION TITLE
HTTPS requests work, but HTTP do not

### DISCUSSION THREAD

**Discussion by lucadenhez** [utmapp/UTM#3021] [UTM Help (macOS)]
**Describe the issue**

In UTM, when I try to access a site via HTTP, (So far tried Ubuntu Server and Kali) it fails and says 'temporary failure in name resolution.' But when I access HTTPS sites, it works perfectly. I've tried shuffling through network emulators in the UTM settings, but no luck. As an alternative I tried using the Parallels VM for Ubuntu, and I can access both HTTP and HTTPS. (I don't like Parallels that much though, I prefer the UTM ecosystem) I asked someone for help on Discord, and they told me to try `dig duckduckgo.com` and `dig 1.1.1.1 duckduckgo.com`, and both commands run successfully. And since they both ran, they told me to file a bug report! 

If the format is incorrect, sorry, these are some of my first bug reports on GitHub. If any information is lacking, please let me know, I'd really like to fix this issue per my gratification and others!

Cheers.

**Configuration:** Ubuntu 20.04 Server ARM64, Kali ARM64
* UTM Version: 2.12 (31)
* OS Version: MacOS 12.0 Beta (Monterey)
* Intel or Apple Silicon? Apple Silicon M1

**VM Config**
[config.plist.zip](https://github.com/utmapp/UTM/files/6987491/config.plist.zip)

**Comment by ty-yqs** [comment]
Because iOS only allows HTTPS requests.

**Comment by lucadenhez** [comment]
I'm on an M1 MacBook Air, running macOS. Not iOS

**Comment by conath** [comment]
Could this be a new security _feature_ in macOS Monterey? I have not seen this issue reported before, and Monterey isn't supported by UTM yet.

**Comment by conath** [comment]
@lucadenhez Thanks for creating this issue. For further investigation, we might need a Debug Log from when the HTTP request fails. Here's how to get the Debug Log:

> **Debug log**
> For all issues, _including_ crashes, you should attach a debug log. Open UTM, and open the settings for the VM you wish to launch. Near the top of the `QEMU` page is `Debug Log`. Turn it on and save the VM. Start the VM. After you reproduce the issue, close the VM, open the VM settings again and select `Export Log...` and attach it here.

**Comment by lucadenhez** [comment]
Sure I will leave one! Thanks for still supporting this issue. So some discoveries: For the past week, I've spent hours just trying and trying commands to fix this issue, and I figured that it is UTM or QEMU, not Ubuntu itself. But today I ran the commands, `sudo dhclient -R` and `sudo dhclient`, and now HTTP and HTTPS request work!! 🎉 Not a Linux pro, so I'm gonna go research on why my DCHP isn't working. I'll sent you a log soon. :)

**Comment by conath** [comment]
Oh nice, in that case the log is not required, as it is not a UTM issue but rather a guest OS issue. Note that I would recommend to use manual DNS setting anyway.

**Reply by osy** [reply]
So is this the same dns issue as before? Still a mystery why that happens. I can’t reproduce it still.

**Reply by conath** [reply]
No it‘s different. I believe the common DNS issue is related to IPv6 networking used by the host. https://github.com/utmapp/UTM/issues/2429

**Comment by lucadenhez** [comment]
Awesome. Yeah I kept trying manual DNS settings earlier, such as 8.8.8.8, etc. But it kept defaulting to 127.0.0.53. (Sounds like loopback; not totally sure) But yeah I hope this issue will resolve itself on other distros! Shall we close this issue?
